src/auth/mas.py

```diff
@@ -63,21 +70,37 @@ def validate_mas_config(container: ops.model.Container) -> None:
 
     Args:
         container: Synapse container.
+
+    Raises:
+        MASConfigInvalidError: when validation of the MAS config failed.
     """
     command = [MAS_EXECUTABLE_PATH, "config", "check", "-c", MAS_CONFIGURATION_PATH]
-    process = container.exec(command=command, working_dir=MAS_WORKING_DIR)
-    process.wait_output()
+
+    try:
+        process = container.exec(command=command, working_dir=MAS_WORKING_DIR)
+        process.wait_output()
+    except ops.pebble.ExecError as exc:
+        logger.exception("Validation of the MAS config failed.")
+        raise MASConfigInvalidError("Validation of the MAS config failed.") from exc
 
 
 def sync_mas_config(container: ops.model.Container) -> None:
     """Sync the MAS configuration with the database.
 
     Args:
         container: Synapse container.
+
+    Raises:
+        MASConfigSyncError: when synchronisation of the MAS config failed.
     """
     command = [MAS_EXECUTABLE_PATH, "config", "sync", "--prune", "-c", MAS_CONFIGURATION_PATH]
-    process = container.exec(command=command, working_dir=MAS_WORKING_DIR)
-    process.wait()
+
+    try:
+        process = container.exec(command=command, working_dir=MAS_WORKING_DIR)
+        process.wait()
+    except ops.pebble.ExecError as exc:
+        logger.exception("Error syncing MAS config with the database.")
+        raise MASConfigSyncError("Error validating MAS configuration.") from exc
```

Comment:
Should the exception description be the same as the logger.exception?

Reply:
thanks, I updated it